core/src/test/java/io/parsingdata/metal/data/ConcatenatedValueSourceTest.java

```diff
@@ -90,4 +100,41 @@ public void checkData(final String description, final int offset, final int leng
         }
     }
 
+    @Test
+    @Timeout(value=1)
+    public void concatenatedValueSourceRead() {
```

Comment:
This test took 1.5 seconds on my computer without the fix, and 0.055 seconds with the fix. :tada: